docs/template_plugin/tests/functional/op_reference/roi_align.cpp

```diff
@@ -0,0 +1,165 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include "openvino/opsets/opset5.hpp"
+#include "openvino/opsets/opset4.hpp"
+#include "openvino/opsets/opset3.hpp"
+#include "openvino/opsets/opset1.hpp"
+#include "base_reference_test.hpp"
+
+using namespace reference_tests;
+using namespace ov;
+
+namespace {
+struct ROIAlignParams {
+    reference_tests::Tensor featureMap;
+    reference_tests::Tensor expectedFeatureMap;
+    reference_tests::Tensor coords;
+    reference_tests::Tensor roiIdx;
+    int32_t pooledH;
+    int32_t pooledW;
+    float spatialScale;
+    int32_t poolingRatio;
+    std::string poolingMode;
+    std::string testcaseName;
+};
+
+struct Builder : ParamsBuilder<ROIAlignParams> {
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, featureMap);
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, expectedFeatureMap);
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, coords);
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, roiIdx);
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, pooledH);
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, pooledW);
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, spatialScale);
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, poolingRatio);
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, poolingMode);
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, testcaseName);
+};
+
+class ReferenceROIAlignTest : public testing::TestWithParam<ROIAlignParams>, public CommonReferenceTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = CreateFunction(params);
+        inputData = {params.featureMap.data};
+        refOutData = {params.expectedFeatureMap.data};
+    }
+
+    static std::string getTestCaseName(const testing::TestParamInfo<ROIAlignParams>& obj) {
+        auto param = obj.param;
+        std::ostringstream result;
+        result << "fType=" << param.featureMap.type;
+        result << "_fShape=" << param.featureMap.shape;
+        result << "_efType=" << param.expectedFeatureMap.type;
+        result << "_efShape=" << param.expectedFeatureMap.shape;
+        result << "_cType=" << param.coords.type;
+        result << "_cShape=" << param.coords.shape;
+        result << "_rType=" << param.roiIdx.type;
+        result << "_rShape=" << param.roiIdx.shape;
+        result << "_pooledH=" << param.pooledH;
+        result << "_pooledW=" << param.pooledW;
+        result << "_spatialScale=" << param.spatialScale;
+        result << "_poolingRatio=" << param.poolingRatio;
+        result << "_poolingMode=" << param.poolingMode;
+        if (param.testcaseName != "") {
+            result << "_=" << param.testcaseName;
+        }
+        return result.str();
+    }
+
+private:
+    static std::shared_ptr<Model> CreateFunction(const ROIAlignParams& params) {
+        const auto featureMap = std::make_shared<opset1::Parameter>(params.featureMap.type, params.featureMap.shape);
```

Comment:
Can you make dynamic shape input for the parameter?

Reply:
PartialShape is applied to input parameter. Thanks for your comment.